File: core/src/main/scala/pl/edu/agh/multiscalemodelling/engine/logic/LogicThread.java
package pl.edu.agh.multiscalemodelling.engine.logic;

import java.util.List;

public class LogicThread extends Thread {

    private List<Cell> cells;

    public LogicThread(List<Cell> cells) {

        super();
        this.cells = cells;

    }

    public void run() {

        for (Cell cell : cells) {
            cell.checkNeighbors();
        }

        for (Cell cell : cells) {
            cell.update();
        }

    }

}

File: core/src/main/scala/pl/edu/agh/multiscalemodelling/engine/logic/LogicThread.scala
package pl.edu.agh.multiscalemodelling.engine.logic

import java.util

import pl.edu.agh.multiscalemodelling.engine.logic.enumeration.{OperationMode, State}
import pl.edu.agh.multiscalemodelling.processsimulation.naiveseedsgrowth.NaiveSeedsGrowthCell

class LogicThread(var cells: util.List[NaiveSeedsGrowthCell]) extends Thread {
  override def run(): Unit = {
    import scala.collection.JavaConversions._
    for (cell <- cells) {
      cell.checkNeighbors(OperationMode.SIMPLE_GROWTH)
    }
    import scala.collection.JavaConversions._
    for (cell <- cells) {
      cell.update()
      if(cell.nextState == State.EMPTY) Logic.allProcessed = false
    }
  }
}